mxtools/flyer.py

```diff
@@ -56,15 +56,21 @@ def describe_collect(self):
                     self.detector.cam.array_size.array_size_y.get(),
                 ],
                 "external": "FILESTORE:",
-            }
+            },
+            "omega": {
+                "source": "read-from-AD-file",
+                "dtype": "array",
+                "shape": [self.detector.cam.num_images.get()],
+            },
         }
         return return_dict
 
     def collect(self):
         self.unstage()
 
         now = ttime.time()
-        data = {f"{self.detector.name}_image": self._datum_ids[0]}
+        self._master_metadata = self._extract_metadata()
+        data = {f"{self.detector.name}_image": self._datum_ids[0], "omega": self._master_metadata}
```

Comment:
Is it possible for this to get called before self._datum_ids gets populated? If so, should we catch this exception?

Reply:
The `_asset_docs_cache` is populated with the `resource` dictionary once the detector is staged, which happens on the `kickoff` step, so it happens much earlier and in the predefined order (`kickoff` -> `complete` -> `collect`. The `self._datum_ids` is populated in the `collect_asset_docs` method, which is called before `collect`, so we should be good here. Also, it's not a new code, it's just the reformatting that caused it to look like a new code after we added the `omega` key.